tests/components/feedreader/test_init.py

```diff
@@ -313,53 +228,89 @@ async def test_feed_without_publication_date_and_title(
-
-        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
-        await hass.async_block_till_done()
+    assert await async_setup_config_entry(
+        hass, VALID_CONFIG_DEFAULT, return_value=feed_three_events
+    )
+    await hass.async_block_till_done()
 
     assert len(events) == 3
 
 
 async def test_feed_with_unrecognized_publication_date(
-    hass: HomeAssistant, events
+    hass: HomeAssistant, events, feed_four_events
 ) -> None:
     """Test simple feed with entry with unrecognized publication date."""
-    with patch(
-        "feedparser.http.get", return_value=load_fixture_bytes("feedreader4.xml")
-    ):
-        assert await async_setup_component(hass, DOMAIN, VALID_CONFIG_2)
-
-        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
-        await hass.async_block_till_done()
+    assert await async_setup_config_entry(
+        hass, VALID_CONFIG_DEFAULT, return_value=feed_four_events
+    )
+    await hass.async_block_till_done()
 
     assert len(events) == 1
 
 
 async def test_feed_invalid_data(hass: HomeAssistant, events) -> None:
     """Test feed with invalid data."""
-    invalid_data = bytes("INVALID DATA", "utf-8")
-    with patch("feedparser.http.get", return_value=invalid_data):
-        assert await async_setup_component(hass, DOMAIN, VALID_CONFIG_2)
-
-        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
-        await hass.async_block_till_done()
+    assert await async_setup_config_entry(
+        hass, VALID_CONFIG_DEFAULT, return_value=bytes("INVALID DATA", "utf-8")
+    )
+    await hass.async_block_till_done()
 
     assert len(events) == 0
 
 
 async def test_feed_parsing_failed(
-    hass: HomeAssistant, events, caplog: pytest.LogCaptureFixture
+    hass: HomeAssistant, events, feed_one_event, caplog: pytest.LogCaptureFixture
 ) -> None:
     """Test feed where parsing fails."""
     assert "Error fetching feed data" not in caplog.text
 
     with patch("feedparser.parse", return_value=None):
-        assert await async_setup_component(hass, DOMAIN, VALID_CONFIG_2)
-
-        hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+        assert not await async_setup_config_entry(
+            hass, VALID_CONFIG_DEFAULT, return_value=feed_one_event
+        )
         await hass.async_block_till_done()
 
     assert "Error fetching feed data" in caplog.text
     assert not events
+
+
+async def test_feed_errors(hass: HomeAssistant, feed_one_event) -> None:
+    """Test feed errors."""
+    entry = MockConfigEntry(domain=DOMAIN, data=VALID_CONFIG_DEFAULT)
+    entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.feedreader.coordinator.feedparser.http.get"
+    ) as feedreader:
+        # success setup
+        feedreader.return_value = feed_one_event
+        assert await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        coordinator: FeedReaderCoordinator = entry.runtime_data
```

Comment:
We should replace this with testing entities once we have entities to test with, we should not test internals